Source/WTF/wtf/threads/Signals.cpp

```diff
@@ -203,6 +204,162 @@ kern_return_t catch_mach_exception_raise_state_identity(mach_port_t, mach_port_t
+    for (mach_msg_type_number_t i = 0; i < threadCount; i++) {
+        if (thread != THREAD_NULL) {
+            // clean up all the unnecessary thread ports like a good citizen
+            ASSERT(thread != threads[i]);
+            mach_port_deallocate(mach_task_self(), threads[i]);
+            continue;
+        }
+
+        kern_return_t kr = thread_info(threads[i], THREAD_IDENTIFIER_INFO,
+        reinterpret_cast<thread_info_t>(&threadIdentifierInfo), &infoCount);
+
+        if (kr == MACH_SEND_INVALID_DEST) {
+            // ignore threads that have been destroyed
+            continue;
+        }
+
+        if (threadIdentifierInfo.thread_id == threadID) {
+            // we found the thread we are looking for
+            thread = threads[i];
+            continue;
+        }
+        mach_port_deallocate(mach_task_self(), threads[i]);
+    }
+    return thread;
+}
+
+kern_return_t catch_mach_exception_raise_state(
+    mach_port_t,
+    exception_type_t,
+    const mach_exception_data_t,
+    mach_msg_type_number_t,
+    int*,
+    const thread_state_t,
+    mach_msg_type_number_t,
+    thread_state_t,
+    mach_msg_type_number_t*)
+{
+    dataLogLn("We should not have called catch_mach_exception_raise_state, please file a bug at bugs.webkit.org");
+    return KERN_FAILURE;
+}
+
+kern_return_t catch_mach_exception_raise_identity_protected(
+    mach_port_t exceptionPort,
+    uint64_t threadID,
+    mach_port_t taskIDToken,
+    exception_type_t exceptionType,
+    mach_exception_data_t exceptionData,
+    mach_msg_type_number_t dataCount)
+{
+    UNUSED_PARAM(taskIDToken);
+
+    SignalHandlers& handlers = g_wtfConfig.signalHandlers;
+    RELEASE_ASSERT(exceptionPort == handlers.exceptionPort);
+    // If we wanted to distinguish between SIGBUS and SIGSEGV for EXC_BAD_ACCESS on Darwin we could do:
+    // if (exceptionData[0] == KERN_INVALID_ADDRESS)
+    //    signal = SIGSEGV;
+    // else
+    //    signal = SIGBUS;
+    Signal signal = fromMachException(exceptionType);
+    RELEASE_ASSERT(signal != Signal::Unknown);
+
+    thread_act_t thread = threadIDGetThreadPort(threadID);
+
+    mach_msg_type_number_t stateCount = 0;
+
+#if CPU(X86_64) || CPU(X86)
+    x86_thread_state_t state;
+    int flavor = x86_THREAD_STATE;
+    stateCount = x86_UNIFIED_THREAD_STATE_COUNT;
+#elif CPU(ARM64) || CPU(ARM)
+    arm_unified_thread_state state;
+    int flavor = ARM_THREAD_STATE;
+    stateCount = ARM_UNIFIED_THREAD_STATE_COUNT;
+#endif
+
+    kern_return_t kr = thread_get_state(thread, flavor, (thread_state_t)&state, &stateCount);
+    if (kr != KERN_SUCCESS) {
+        dataLogLn("thread_get_state failed due to ", mach_error_string(kr));
+        return kr;
+    }
+
+#if CPU(ARM64E) && OS(DARWIN)
+    ptrauth_generic_signature_t inStateHash = hashThreadState((thread_state_t)&state);
+#endif
+
+#if CPU(X86_64)
+    PlatformRegisters& registers = state.uts.ts64;
+    ASSERT(state.ash.flavor == X86_THREAD_STATE64);
```

Comment:
Is there a reason you converted this into a Debug only `ASSERT`?  If not, let's make this a `RELEASE_ASSERT` to match prior behavior?